backend/routes/users.js

```diff
@@ -2,41 +2,70 @@
 const express = require('express');
 const { check, oneOf, validationResult } = require('express-validator/check');
 const { matchedData } = require('express-validator/filter');
+const bcrypt = require('bcrypt');
 const router = express.Router();
 
 // Local Imports
 const User = require('../models/user');
 
-router.route('/')
-  .get((req, res) => {
+
+// This method is purposely put before the login wall so that
+// new users can be created w/o needing to be signed in...
+router.post('/', [
+  check('first_name').isAlpha().trim().escape(),
+  check('password').isAscii().trim().escape(),
+  check('last_name').isAlpha().trim().escape(),
+  check('email').isEmail().trim(),
+  check('street_address').isAscii().trim().escape(),
+  check('city').isAlpha().trim().escape(),
+  check('state').isAlpha().trim().escape(),
+  check('zip_code').isAscii().trim().escape(),
+  check('phone_number').isAscii().trim().escape(),
+  check('date_of_birth').exists().trim().escape()
+], (req, res) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.mapped() });
+  }
+  const userData = matchedData(req);
+  let emailInUse = false;
+  User.findOne({ email: userData.email })
+    .then(user => {
+      if (user) {
+        emailInUse = true;
+        throw new Error("Email already in use")
```

Comment:
Feels somewhat janky but I'm okay with it

Reply:
I wasn't sure how to tell from the Error message in the catch block what caused the error. I only wanna tell them that their email is in use but no other error message info... that's how I thought of doing it.. Suggestions?